payloads/edk2/Kconfig.dasharo: bump edk2 revision

aronowski opened the pull request:
~Update the testing workflow with the OSFV test suite to prove, that VirtIO drivers are visible in QEMU. Update and install the required tools to set up the test suite with osfv-test-data. Use the edk2 revision with support for building VirtIO drivers.~

~See https://github.com/Dasharo/dasharo-issues/issues/901#issuecomment-3005462400 for context. After https://github.com/Dasharo/edk2/pull/258 is merged, update `CONFIG_EDK2_REPOSITORY` and `CONFIG_EDK2_TAG_OR_REV`. After https://github.com/Dasharo/open-source-firmware-validation/pull/905 is merged, update the `Checkout OSFV repository and submodules` job.~

---

Bump edk2 revision, which ships VirtIO storage drivers for q35 mainboard. [A run that proves, that it builds fine](https://github.com/aronowski/coreboot/actions/runs/16099284577/job/45426275191).

philipanda commented:
The PR looks good overall. We can merge it after the https://github.com/Dasharo/edk2/pull/258 and after changing all of the references to your forks.

Please update your commit messages according to the [check-upstream-status](https://github.com/Dasharo/coreboot/actions/runs/15890146515/job/44825548126?pr=704) CI check. Here are some docs on how to add it https://docs.dasharo.com/dev-proc/source-code-structure/#upstream-status

Do you have an idea why [this check](https://github.com/Dasharo/coreboot/actions/runs/15890146525/job/44826880940?pr=704) fails? The cloned repository is private and a token is required to access it. 
[It is not provided in this workflow](https://github.com/Dasharo/coreboot/actions/runs/15890146525/job/44826880940?pr=704#step:5:18) despite it being there in [any other PR originating from this repo](https://github.com/Dasharo/coreboot/actions/runs/15883233822/job/44788789347#step:5:6), not from a fork.
Do you think we could fix it to make all CI tests pass?

aronowski commented:
> Do you have an idea why [this check](https://github.com/Dasharo/coreboot/actions/runs/15890146525/job/44826880940?pr=704) fails? The cloned repository is private and a token is required to access it. [It is not provided in this workflow](https://github.com/Dasharo/coreboot/actions/runs/15890146525/job/44826880940?pr=704#step:5:18) despite it being there in [any other PR originating from this repo](https://github.com/Dasharo/coreboot/actions/runs/15883233822/job/44788789347#step:5:6), not from a fork. Do you think we could fix it to make all CI tests pass?

Seems like a reasonable security measure on GitHub's end, so that only those in the Dasharo organization can access the token in their running workflows. Otherwise any malicious actor could file a PR with a script, that emails them the token (or any other secrets).

If this hypothesis is true, then fixing it would require inviting me to the [Dasharo/coreboot](https://github.com/Dasharo/coreboot) repository with the minimal permissions needed for the token to show up in my running workflows. Otherwise I'd bet that the issue will fix itself after merging.

philipanda commented:
> Seems like a reasonable security measure on GitHub's end, so that only those in the Dasharo organization can access the token in their running workflows. Otherwise any malicious actor could file a PR with a script, that emails them the token (or any other secrets).

Without the permissions you can't run the workflows at all. They were authorised by me and that's why I find it unintuitive that the token was not released. The workflows run on an external GH runner and the token is not accessible to anyone looking at the results so I don't understand how that is supposed to help with security.

Anyway, it might be worth creating an issue for that if there is no obvious fix. There will be more contributions from people outside of the organisation and it would be great to have the workflows at least inform that the token cannot be used instead of simply failing.

Would you like to try and create an issue for that? It doesn't need to be very verbose unless you have some ideas on how to handle that. If not, I'll make one.

aronowski commented:
> The workflows run on an external GH runner and the token is not accessible to anyone looking at the results so I don't understand how that is supposed to help with security.

I can imagine that if the token can be used inside of the actual workflow (**not** when viewing logs or browsing a workspace) to clone a private repository, it could well be sent in plaintext to an attacker.

> Would you like to try and create an issue for that? It doesn't need to be very verbose unless you have some ideas on how to handle that.

I can create one after some basic experiments to suggest an optimal solution - e.g. if a conditional in a workflow would be OK, or if I should think of something else.

philipanda commented:
> I can imagine that if the token can be used inside of the actual workflow (not when viewing logs or browsing a workspace) to clone a private repository, it could well be sent in plaintext to an attacker.

That's why it has to be authorised, right? With the exact same priviledges a change in the workflow from an attacker can be merged into the upstream and also leak the tokens. Anyway

> I can create one after some basic experiments to suggest an optimal solution - e.g. if a conditional in a workflow would be OK, or if I should think of something else.

It's not in any way required for the bounty, that's why I asked. If you are willing to investigate then that's great.

philipanda commented on .github/workflows/test.yml:
```diff
@@ -7,27 +7,38 @@ on:
-           sudo apt-get install qemu-system-x86-64 swtpm
+           sudo apt-get install qemu-system-x86-64 swtpm git-annex
+
+      # Pulling secure-boot/sb_test_data.img fails, carry on anyway
```
Yes, [there's an issue for that,](https://github.com/Dasharo/osfv-test-data/issues/11) it's expected

aronowski commented:
Changed the edk2 repository reference from my fork to Dasharo/edk2 and [the current revision builds fine](https://github.com/aronowski/coreboot/actions/runs/16078076279/job/45377643551).

Should I wait for https://github.com/Dasharo/open-source-firmware-validation/pull/905 to be merged, or change the testing workflow's OSFV references to either
```
repository: 'Dasharo/open-source-firmware-validation'
ref: 'develop'
```
or
```
repository: 'Dasharo/open-source-firmware-validation'
ref: 'qemu-lib-refactor'
```
before setting this PR as ready for review?

philipanda commented:
> Changed the edk2 repository reference from my fork to Dasharo/edk2 and [the current revision builds fine](https://github.com/aronowski/coreboot/actions/runs/16078076279/job/45377643551).
> 
> Should I wait for [Dasharo/open-source-firmware-validation#905](https://github.com/Dasharo/open-source-firmware-validation/pull/905) to be merged, or change the testing workflow's OSFV references to either
> 
> ```
> repository: 'Dasharo/open-source-firmware-validation'
> ref: 'develop'
> ```
> 
> or
> 
> ```
> repository: 'Dasharo/open-source-firmware-validation'
> ref: 'qemu-lib-refactor'
> ```
> 
> before setting this PR as ready for review?

We can't merge the changes to OSFV right now, because the drives would be unusable. A QEMU release must first be published.

I think we should move the workflow to a separate PR, that we will merge later once everything works. Updating the EDK2 revisions is the most important change here. Also remember to rebase onto the newest `dasharo` branch.

aronowski committed:
payloads/edk2/Kconfig.dasharo: bump edk2 revision

This revision ships VirtIO storage drivers for q35 mainboard

Upstream-Status: Inappropriate [Dasharo downstream]
Signed-off-by: Kamil Aronowski <kamil.aronowski@yahoo.com>

aronowski commented:
> Also remember to rebase onto the newest `dasharo` branch.

Appreciate the hints!

> I think we should move the workflow to a separate PR, that we will merge later once everything works. Updating the EDK2 revisions is the most important change here.

Updated the reference - [a run that proves, that it builds fine](https://github.com/aronowski/coreboot/actions/runs/16099284577/job/45426275191).
Once it's merged, I will, as suggested, open another PR with the testing workflow.

macpijan commented:
>  Once it's merged, I will, as suggested, open another PR with the testing workflow.

Thanks. Merged that and looking for a follow-up then.
